esphome/components/ld2420/ld2420.cpp

```diff
@@ -0,0 +1,506 @@
+              sizeof(this->new_config_.low_thresh))) != 0 ||
+      (memcmp(&this->new_config_.min_gate, &this->current_config_.min_gate, 0x6)) != 0) {
+    ESP_LOGD(TAG, "Config changed, writing update to LD2420.");
+    this->set_min_max_distances_timeout_(this->new_config_.max_gate, this->new_config_.min_gate,
+                                         this->new_config_.timeout);
+    for (uint8_t gate = 0; gate < 16; gate++) {
+      this->set_gate_threshold_(gate);
+    }
+    restart = true;
+  }
+
+  this->set_system_mode_(CMD_SYSTEM_MODE_NORMAL);
+  this->set_config_mode_(false);
+  if (restart)
+    this->restart_();
+  ESP_LOGCONFIG(TAG, "LD2420 setup complete.");
+}
+
+void LD2420Component::loop() {
+  // If there is a active send command do not process it here, the send command call will handle it.
+  if (!get_cmd_active_()) {
+    if (!available())
+      return;
+    static uint8_t buffer[2048];
+    static uint8_t rx_data;
+    while (available()) {
+      rx_data = read();
+      this->readline_(rx_data, buffer, sizeof(buffer));
+    }
+  }
+}
+
+/*
+Configure commands - little endian
+
+No command can exceed 64 bytes, otherwise they would need be to be split up into multiple sends.
+
+All send command frames will have:
+  Header = FD FC FB FA, Bytes 0 - 3, uint32_t 0xFAFBFCFD
+  Length, bytes 4 - 5, uint16_t 0x0002, must be at least 2 for the command byte if no addon data.
+  Command bytes 6 - 7, uint16_t
+  Footer = 04 03 02 01 - uint32_t 0x01020304, Always last 4 Bytes.
+Receive
+  Error bytes 8-9 uint16_t, 0 = success, all other positive values = error
+
+Enable config mode:
+Send:
+  UART Tx: FD FC FB FA 04 00 FF 00 02 00 04 03 02 01
+  Command = FF 00 - uint16_t 0x00FF
+  Protocol version = 02 00, can be 1 or 2 - uint16_t 0x0002
+Reply:
+  UART Rx: FD FC FB FA 06 00 FF 01 00 00 02 00 04 03 02 01
+
+Disable config mode:
+Send:
+  UART Tx: FD FC FB FA 02 00 FE 00 04 03 02 01
+  Command = FE 00 - uint16_t 0x00FE
+Receive:
+  UART Rx: FD FC FB FA 04 00 FE 01 00 00 04 03 02 01
+
+Configure system parameters:
+UART Tx: FD FC FB FA 08 00 12 00 00 00 64 00 00 00 04 03 02 01  Set system parms
+Command = 12 00 - uint16_t 0x0012
+
+Configure gate sensitivity parameters:
+UART Tx: FD FC FB FA 0E 00 07 00 10 00 60 EA 00 00 20 00 60 EA 00 00 04 03 02 01
+Command = 12 00 - uint16_t 0x0007
+Gate 0 high thresh = 10 00 uint16_t 0x0010, Threshold value = 60 EA 00 00 uint32_t 0x0000EA60
+Gate 0 low thresh = 20 00 uint16_t 0x0020, Threshold value = 60 EA 00 00 uint32_t 0x0000EA60
+
+*/
+
+void LD2420Component::readline_(uint8_t rx_data, uint8_t *buffer, int len) {
+  static int pos = 0;
+
+  if (rx_data >= 0) {
+    if (pos < len - 1) {
+      buffer[pos++] = rx_data;
+      buffer[pos] = 0;
+    } else {
+      pos = 0;
+    }
+    if (pos >= 4) {
+      if (memcmp(&buffer[pos - 4], &CMD_FRAME_FOOTER, sizeof(CMD_FRAME_FOOTER)) == 0) {
+        ESP_LOGD(TAG, "Rx: %s", format_hex_pretty(buffer, pos).c_str());
```

Comment:
Let's maybe make this verbose or even very-verbose along with those below -- otherwise it really makes the logs unnecessarily noisy.

```suggestion
        ESP_LOGVV(TAG, "Rx: %s", format_hex_pretty(buffer, pos).c_str());
```